keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/BaseClusterHandler.kt

```diff
@@ -0,0 +1,51 @@
+package com.netflix.spinnaker.keel.api.plugins
+
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceDiff
+import com.netflix.spinnaker.keel.api.ResourceSpec
+import com.netflix.spinnaker.keel.diff.toIndividualDiffs
+
+/**
+ * Common cluster functionality.
+ * This class contains abstracted logic where we want all clusters to make the
+ * same decision, or abstracted logic involving coordinating resources or
+ * publishing events.
+ */
+abstract class BaseClusterHandler<SPEC: ResourceSpec, RESOLVED: Any>(
+  resolvers: List<Resolver<*>>
+) : ResolvableResourceHandler<SPEC, Map<String, RESOLVED>>(resolvers) {
+
+  abstract fun getDesiredRegion(diff: ResourceDiff<RESOLVED>): String
```

Comment:
Consider adding doc comments for the functions in the base class since they'll be reused/overridden.